File: CS/CS.NET-VB.NET/FormToForm/VB2008FormToFormProperty/VB2008FormToFormProperty/Form2.vb
Public Class Form2

    Dim F1 As Form1 '

    ' Dim F1 As Form1 = CType(Application.OpenForms("Form1"), Form1) 

    Public Sub New() '

        ' This call is required by the Windows Form Designer.
        InitializeComponent() '

        ' Add any initialization after the InitializeComponent() call.

    End Sub '

    '
    Public Sub New(ByVal F1 As Form1)
        Me.New()
        Me.F1 = F1
    End Sub
    '

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        F1.TextBoxText = TextBox1.Text
        Me.Close()

    End Sub

    Private Sub Form2_Shown(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Shown

        TextBox1.Text = F1.TextBoxText
        TextBox1.Focus()

    End Sub

End Class

File: CS/CS.NET/Windows Forms .NET/FormToForm/Property/2/CS2008FormToFormProperty/CS2008FormToFormProperty/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace CS2008FormToFormProperty
{
    public partial class Form2 : Form
    {
        Form1 F1; 

        /*
        public Form2()
        {
            InitializeComponent();
        }

        public Form2(Form1 F1) : this()
        {
            this.F1 = F1;
        }   
        */

        public Form2(Form1 F1) // : base()
        {
            InitializeComponent();
            this.F1 = F1;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            F1.TextBoxText = textBox1.Text;
            this.Close();
        }

        private void Form2_Shown(object sender, EventArgs e)
        {
            textBox1.Text = F1.TextBoxText;
            textBox1.Focus();
        }
    }
}
